From d02e89859f5a09ff0a3fae01438b63a4f11c8619 Mon Sep 17 00:00:00 2001
From: Victor Axelrod <victora@marvell.com>
Date: Mon, 11 Apr 2016 14:30:54 +0300
Subject: [PATCH 101/239] mv_ddr: apn806: Route DDR traffic through the
 McKinley6 controller

Improve DDR performance by using the McKinley6 Memory controller

Change-Id: Iba66fb4c7043578bda8f6be091d185740352d4aa
Signed-off-by: Victor Axelrod <victora@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/28922
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Nadav Haklai <nadavh@marvell.com>
---
 drivers/marvell/mv_ddr/apn806/mv_ddr_apn806.c      |  3 +
 .../marvell/mv_ddr/apn806/mv_ddr_apn806_static.h   | 90 ++++++++++++++--------
 2 files changed, 62 insertions(+), 31 deletions(-)

diff --git a/drivers/marvell/mv_ddr/apn806/mv_ddr_apn806.c b/drivers/marvell/mv_ddr/apn806/mv_ddr_apn806.c
index 84777e8..e820c68 100644
--- a/drivers/marvell/mv_ddr/apn806/mv_ddr_apn806.c
+++ b/drivers/marvell/mv_ddr/apn806/mv_ddr_apn806.c
@@ -616,6 +616,9 @@ int mv_ddr_pre_training_soc_config(const char *ddr_type)
 
 int mv_ddr_post_training_soc_config(const char *ddr_type)
 {
+	/* set mux to MC6 */
+	mmio_write_32(0xf00116d8, 0x38c);
+
 	return MV_OK;
 }
 
diff --git a/drivers/marvell/mv_ddr/apn806/mv_ddr_apn806_static.h b/drivers/marvell/mv_ddr/apn806/mv_ddr_apn806_static.h
index 85fc490..4b2fa1b 100644
--- a/drivers/marvell/mv_ddr/apn806/mv_ddr_apn806_static.h
+++ b/drivers/marvell/mv_ddr/apn806/mv_ddr_apn806_static.h
@@ -147,12 +147,6 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #define DRAM_DIRECT_START		(0x6000000)
 #define DRAM_DIRECT_SIZE		(2048)
 
-/*
- * CONFIG_DUNIT_STATIC for static configuration thru TIP,
- * otherwise, thru MK6.
- */
-#define CONFIG_DUNIT_STATIC
-
 struct mk6_reg_data {
 	uint32_t offset;
 	uint32_t value;
@@ -2251,34 +2245,42 @@ struct mk6_reg_data ddr_static_setup[] = {
 #else /* ddr static configuratioin thru mk6 */
 struct mk6_reg_data ddr_static_setup[] = {
 	/* Traffic thru Mckinley 6::MC init for MISL BOARD 2CS 4Gb x8 devices of micron - 2133P */
-	{0xF00116D8, 0x38A},
-	{0xF06F0100, 0x44C0000},
+	{0xF00116D8, 0x3CC},
+	{0xF06F0100, 0x44C0006},
 	{0xF00119D4, 0x2},
-	{0xF06F8D36, 0xACAC0000},
+	{0xF06F8D38, 0xACAC0000},
 	{0xF06F4360, 0xFFFF0044},
-	{0xF00114C8, 0x5145941},
-	{0xF00117C8, 0x15145941},
-	{0xF0011DC8, 0x5145941},
-	{0xF0011EC8, 0x15145941},
-	{0xF00114CC, 0x12015},
+	{0xF00114C8, 0x1840008},
+	{0xF00117C8, 0x28A0008},
+	{0xF0011DC8, 0x1840008},
+	{0xF0011EC8, 0x28A0008},
+	{0xf00116a0, 0xE83F0001},
+	{0xF00114CC, 0x1200D},
+	{0xF0011524, 0x8800},
+	{0xF0011400, 0x6b104c30},
+	{0xF0011400, 0x7b104c30},
 	{0xF0011538, 0xD0D},
-	{0xF001153C, 0x1313},
-	{0xF0020064, 0x15},	/* MC"readReady"+MC2PHYlatency */
-	{0xF0021180, 0x200},
-	{0xF0021000, 0xC0},	/* phy_rfifo_rptr_dly_val */
+	{0xF001153C, 0x1515},
+
+	{0xF0020064, 0x606},		/* MC "readReady"+ MC2PHYlatency */
+	{0xF0021180, 0x500},
+	{0xF0021000, 0x60},		/* phy_rfifo_rptr_dly_val */
+	{0xF00210c0, 0x81000001},
+	{0xF00202c8, 0xfefe},
+
 	{0xf00116a0, 0xcc000000},	/* CKdelayADLL */
-	{0xf00116a0, 0xc0000047},	/* WLpup0CS0 */
-	{0xf00116a0, 0xc0400051},	/* WLpup1 */
-	{0xf00116a0, 0xc0800013},	/* WLpup2 */
-	{0xf00116a0, 0xc0c0000E},	/* WLpup3 */
-	{0xf00116a0, 0xc002004E},	/* RLpup0 */
-	{0xf00116a0, 0xc042004F},	/* RLpup1 */
-	{0xf00116a0, 0xc0820018},	/* RLpup2 */
-	{0xf00116a0, 0xc0c20048},	/* RLpup3 */
-	{0xf00116a0, 0xc0010011},	/* Ctxpup0 */
-	{0xf00116a0, 0xc041001B},	/* Ctxpup1 */
-	{0xf00116a0, 0xc081001D},	/* Ctxpup2 */
-	{0xf00116a0, 0xc0c10018},	/* Ctxpup3 */
+	{0xf00116a0, 0xc0009c5f},
+	{0xf00116a0, 0xc0405c8f},
+	{0xf00116a0, 0xc0804c4b},
+	{0xf00116a0, 0xc0c02040},
+	{0xf00116a0, 0xc0020108},
+	{0xf00116a0, 0xc0420109},
+	{0xf00116a0, 0xc08200Cf},
+	{0xf00116a0, 0xc0c200de},
+	{0xf00116a0, 0xc0030007},
+	{0xf00116a0, 0xc043000A},
+	{0xf00116a0, 0xc0830008},
+	{0xf00116a0, 0xc0c30008},
 	{0xf00116a0, 0xc0040044},	/* WLpup0CS1 */
 	{0xf00116a0, 0xc044004E},	/* WLpup1 */
 	{0xf00116a0, 0xc0840010},	/* WLpup2 */
@@ -2302,6 +2304,32 @@ struct mk6_reg_data ddr_static_setup[] = {
 	{0xf00116a0, 0xEC24070C},	/* ctrldrivestrengthBC */
 	{0xf00116a0, 0xE824070E},	/* datadrivestrengthBC */
 	{0xf00116a0, 0xE8260040},	/* dataODTBC */
+
+	{0xF0011400, 0x7b104c30},
+	{0xF0011404, 0x36301848},
+	{0xF0011408, 0x5411cbb9},
+	{0xF001140c, 0x784A1f97},
+	{0xF0011410, 0x17114444},
+	{0xF0011414, 0x7c8},
+	{0xF0011424, 0x60f3f7},
+	{0xF001142c, 0x14c5138},
+	{0xF0011498, 0xf},
+	{0xF001149c, 0x303},
+	{0xF00114e0, 0x400},
+	{0xF00114e4, 0x3ff},
+	{0xF00114e8, 0x7f01ff},
+	{0xF00114ec, 0x54},
+	{0xF00115e0, 0x23},
+	{0xF00115e4, 0x203c18},
+	{0xF00115ec, 0xd9ff0029},
+	{0xF0011900, 0x314},		/* MR0 */
+	{0xF0011904, 0x001},		/* MR1 */
+	{0xF0011908, 0x200},		/* MR2 */
+	{0xF001190c, 0x800},		/* MR3 */
+	{0xF0011910, 0x0},		/* MR4 */
+	{0xF0011914, 0x4A0},		/* MR5 */
+	{0xF0011918, 0x80A},		/* MR6 */
+
 	{0xf0020340, 0x0},
 	{0xf0020344, 0x30000000},
 	{0xf0020310, 0x21000000},
@@ -2354,7 +2382,7 @@ struct mk6_reg_data ddr_static_setup[] = {
 	{0xf00203c0, 0x10400},
 	{0xf00203c4, 0x0},
 	{0xf00203cc, 0xf010345},
-	{0xF0011480, 0x1},	/* Tipinit */
+	{0xF0011480, 0x1},		/* Tipinit */
 	{0xF0020020, 0x13000001},	/* MCinit */
 	{-1, -1},
 };
-- 
1.9.1

